java-util/src/main/java/io/druid/java/util/common/guava/ExecuteWhenDoneYielder.java

```diff
@@ -56,9 +56,10 @@ public boolean isDone()
   @Override
   public void close() throws IOException
   {
-    if (isDone()) {
+    boolean done = isDone();
+    baseYielder.close();
```

Comment:
It looks like Sequences.withEffect is only used for populating cache (in CachingClusteredClient and CachingQueryRunner). In that case, I think it's important to avoid populating the cache if the sequence isn't done, since the cache should only contain full result sets.

I'm not sure if there's a good reason that limited sequences have that isDone behavior. I would think a limited sequence that reaches its limit is "done". But maybe there's something I'm missing.